File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        for(int i=0; i<n; i++) {
            if(visited[i]) continue;
            
            answer++;
            
            Deque<Integer> dq = new ArrayDeque<>();
            dq.add(i);
            
            while(!dq.isEmpty()) {
                int now = dq.poll();
                
                for(int j = 0; j<n; j++) {
                    if(computers[now][j]==1 && !visited[j]) {
                        visited[j] = true;
                        dq.add(j);
                    }
                }
            }
            
        }
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    #   1  2  3
    # 1[1, 1, 0], 
    # 2[1, 1, 0], 
    # 3[0, 0, 1]

    cnt = 0
    visited = [0] * n
    
    for i in range(n):
        
        if visited[i] == 1 :
            continue
            
        cnt += 1    
        dq = deque()
        dq.append(i)
        visited[i] = 1
        
        
        while dq:
            node = dq.popleft()

            for j in range(n): 
                if computers[node][j] == 1 and visited[j] == 0:
                    visited[j] = 1
                    dq.append(j)
                    
    return cnt